inference-engine/src/gna_plugin/gna_graph_tools.hpp

```diff
@@ -436,7 +436,45 @@ inline void CNNNetSwapLayers(InferenceEngine::CNNLayerPtr lhs,
     lhs->outData.front()->setDims(rhs->outData.front()->getDims());
 }
 
+/**
+* @brief changes the Tensor Desctiption if data by created a new one with correct description and replacing original one
+*/
+inline DataPtr CNNDataChangeTensorDescription(DataPtr data, TensorDesc td) {
+    auto new_dataPtr = std::make_shared<Data>(data->getName() + "_reshaped", td);
+    getInputTo(new_dataPtr) = getInputTo(data);
+    auto creatorLayer = getCreatorLayer(data).lock();
+    getCreatorLayer(new_dataPtr) = creatorLayer;
+
+    auto idx = -1;
+    for (int i=0; i < creatorLayer->outData.size(); i++) {
+        if (areEqualDatas(data, creatorLayer->outData[i])) idx = i;
+    }
+    if (idx == -1) THROW_GNA_EXCEPTION << "No idx for data was found";
+
+    creatorLayer->outData[idx] = new_dataPtr;
+    auto input_to = getInputTo(new_dataPtr);
+    for (auto input : input_to) {
+        for (int input_idx = 0; input_idx < input.second->insData.size(); input_idx++) {
+            if (areEqualDatas(input.second->insData[input_idx].lock(), data)) {
```

Comment:
If you use CNNLayerFindInsDataIdxes, it will return indices and you will not have to do comparison for second time.